[Bug] Unable to install in Fedora Wayland

### Description

They do not work as expected. There is no `build-binary` task. Installing espanso is easier with Terra, as shown [here](https://github.com/espanso/espanso/issues/1948). 

This is a bug because it wasted thirty minutes of my time, and the docs do not work as expected. 

### Steps to reproduce

1. Try to install espanso on Wayland (in my case, Fedora)
2. Fail at the `build-binary` part.

### Expected behavior

The installation docs should be useful. 

### Screenshots

_No response_

### Logs

_No response_

### Your environment

- Public source code:
- Environment name and version (e.g., Rust 1.59.0):
- Operating system and version (e.g., Ubuntu 20.04.2 LTS):
- Espanso version (e.g., 2.1.4-beta):


### Self-service

- [ ] I'd be willing to fix this bug myself.

Hey just found this. I believe similar / same issue as OP just on openSUSE Tumbleweed: `[cargo-make][1] ERROR - Task build-binary not found` 

When trying this

> ```
> cargo build --release --no-default-features --features wayland,modulo,vendored-tls
> ```

I get:
<details><summary>shell output</summary>
<p>

```shell
   Compiling rayon v1.5.3
   Compiling html2text v0.12.0
   Compiling smithay-client-toolkit v0.18.1
warning: espanso-detect@0.1.0: src/evdev/native.cpp:50:10: fatal error: xkbcommon/xkbcommon.h: No such file or directory
warning: espanso-detect@0.1.0:    50 | #include "xkbcommon/xkbcommon.h"
warning: espanso-detect@0.1.0:       |          ^~~~~~~~~~~~~~~~~~~~~~~
warning: espanso-detect@0.1.0: compilation terminated.
error: failed to run custom build command for `espanso-detect v0.1.0 (/home/xxx/espanso/espanso-detect)`

Caused by:
  process didn't exit successfully: `/home/xxx/espanso/target/release/build/espanso-detect-846903fc8dfef6b3/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-changed=src/x11/native.cpp
  cargo:rerun-if-changed=src/x11/native.h
  cargo:rerun-if-changed=src/evdev/native.cpp
  cargo:rerun-if-changed=src/evdev/native.h
  TARGET = Some("x86_64-unknown-linux-gnu")
  OPT_LEVEL = Some("3")
  HOST = Some("x86_64-unknown-linux-gnu")
  CXX_x86_64-unknown-linux-gnu = None
  CXX_x86_64_unknown_linux_gnu = None
  HOST_CXX = None
  CXX = None
  CXXFLAGS_x86_64-unknown-linux-gnu = None
  CXXFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CXXFLAGS = None
  CXXFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "src/evdev" "-Wall" "-Wextra" "-o" "/home/xxx/espanso/target/release/build/espanso-detect-1551bfc6c57598fc/out/src/evdev/native.o" "-c" "src/evdev/native.cpp"
  cargo:warning=src/evdev/native.cpp:50:10: fatal error: xkbcommon/xkbcommon.h: No such file or directory
  cargo:warning=   50 | #include "xkbcommon/xkbcommon.h"
  cargo:warning=      |          ^~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=compilation terminated.
  exit status: 1

  --- stderr


  error occurred: Command "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "src/evdev" "-Wall" "-Wextra" "-o" "/home/xxx/espanso/target/release/build/espanso-detect-1551bfc6c57598fc/out/src/evdev/native.o" "-c" "src/evdev/native.cpp" with args "c++" did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
```

</p>
</details> 




Help would be much appreciated as the Espanso version in the openSUSE Tumbleweed package repos only works for X11 sessions / windows which I'd like to avoid.

--- a/docs/src/ch03-03-linux.md
+++ b/docs/src/ch03-03-linux.md
@@ -1,7 +1,19 @@
 # linux
 
 Espanso on Linux comes in two different flavors: one for X11 and one for Wayland.
-If you don't know which one to choose, follow [these steps to determine which one you are running](https://unix.stackexchange.com/a/325972).
+The variant you build or install **must match your active session**, otherwise Espanso may install but silently fail to work.
+
+#### Determining your session type
+
+Run the following command to check whether you are running X11 or Wayland:
+
+```console
+echo $XDG_SESSION_TYPE
+```
+
+This prints either `x11` or `wayland`. If the variable is empty, follow [these steps to determine which one you are running](https://unix.stackexchange.com/a/325972).
+
+If you are on Wayland (the default on Fedora and recent GNOME-based distributions), build or install the Wayland variant. If you are on X11, use the X11 variant.
 
 #### Necessary dependencies
 

